### PR TITLE
Import paths #2733

### LINKED ISSUE #2733: Import Request in the Template file
I have been creating many spiders recently, and I noticed I had to add below line every time

`from scrapy.http import Request`

I think for most cases we need the Request to crawl to other pages, only for very simple spider we don't need that. Is it possible to include that line to the template?

### PR DESCRIPTION
**Pull request by marlenachatzigrigoriou**
@Gallaecio please check this file and let me know if this is the way that I should carry this issue, so I will be able to continue with the other files.
Fixes #2733.

### CODE REVIEW

**Comment by Gallaecio** [comment]
That’s the idea. However, mind that this pull request includes unrelated commits that you probably dragged from another pull request.

File: docs/topics/request-response.rst [hunk 1]
```diff
@@ -488,7 +488,10 @@ fields with form data from :class:`Response` objects.
 
 .. _lxml.html forms: https://lxml.de/lxmlhtml.html#forms
 
-.. class:: FormRequest(url, [formdata, ...])
+.. class:: scrapy.http.request.form.FormRequest
+.. class:: scrapy.http.request.FormRequest
```
Comment (Gallaecio) [inline docs/topics/request-response.rst:492]: This import path does not actually exist, only the other 3 do.

Comment (marlenachatzigrigoriou) [reply docs/topics/request-response.rst:492]: Do you mean line 491?

Comment (Gallaecio) [reply docs/topics/request-response.rst:492]: No, I mean line 492. `scrapy.http.request.form.FormRequest` [exists](https://github.com/scrapy/scrapy/blob/master/scrapy/http/request/form.py#L19), but `scrapy.http.request.FormRequest` [does not](https://github.com/scrapy/scrapy/blob/master/scrapy/http/request/__init__.py).

Comment (marlenachatzigrigoriou) [reply docs/topics/request-response.rst:492]: I see. Great, thank you!

**Comment by Gallaecio** [comment]
Could you find out why the pull request diff shows so many changes? I suspect you might have accidentally changed the line ending of the files you edited.

**Comment by marlenachatzigrigoriou** [comment]
You are right @Gallaecio. I had accidentally changed the `autocrlf` configuration. Should I delete this commit and create a new one?

**Comment by Gallaecio** [comment]
Either that or simply create a new one. In pull requests, at least in Scrapy, we are not strict at all with commit cleanliness, and in fact _not_ rebasing (and hence force-pushing) can make it easier for reviewers.

File: docs/topics/items.rst [hunk 2]
```diff
@@ -42,7 +42,8 @@ Item objects
 :class:`Item` provides a :class:`dict`-like API plus additional features that
 make it the most feature-complete item type:
 
-.. class:: Item([arg])
+.. class:: scrapy.item.Item
+.. class:: scrapy.Item([arg])
```
Comment (Gallaecio) [inline docs/topics/items.rst:46]: When not using `autoclass`, the signature needs to be defined manually:

```suggestion
.. class:: scrapy.item.Item([arg])
.. class:: scrapy.Item([arg])
```

Same applies to other, similar changes below.

File: docs/topics/practices.rst [hunk 3]
```diff
@@ -1,264 +1,264 @@
-.. _topics-practices:
```
Comment (Gallaecio) [inline docs/topics/practices.rst:1]: It looks like in this file you had the line-ending issue. It also affects the `spiders.rst` file.

Git usually handles this automatically, maybe there’s something you can do in your Git configuration to have Git send the changes with the expected line ending even if locally you use a different line ending.

**Review by Gallaecio — COMMENTED** [review]
Great job, it’s getting there!

File: docs/topics/items.rst (hunk 2 above)
Comment (marlenachatzigrigoriou) [reply docs/topics/items.rst:46]: Thank you, I will fix it!

File: docs/topics/practices.rst (hunk 3 above)
Comment (marlenachatzigrigoriou) [reply docs/topics/practices.rst:1]: I fixed the line-ending of `spiders.rst`. Also, `practices.rst` has no changes (I don't know it was pulled) so don't worry about reviewing it.

**Comment by marlenachatzigrigoriou** [comment]
@Gallaecio the rest of the documentation files that require changes, will be committed by my colleague @mmitropoulou.

File: docs/topics/practices.rst (hunk 3 above)
Comment (Gallaecio) [reply docs/topics/practices.rst:1]: Be aware that both files still show up in the pull request with the line-ending change.

File: docs/news.rst [hunk 4]
```diff
@@ -394,16 +394,16 @@ New features
...
     :class:`response.request <scrapy.http.Response.request>`:
 
     -   The response is handled by the callback of that custom
-        :class:`~scrapy.http.Request` object, instead of being handled by the
-        callback of the original :class:`~scrapy.http.Request` object
+        :class:`~scrapy.Request` object, instead of being handled by the
+        callback of the original :class:`~scrapy.Request` object
 
-    -   That custom :class:`~scrapy.http.Request` object is now sent as the
+    -   That custom :class:`~scrapy..Request` object is now sent as the
```
Comment (elacuesta) [inline docs/news.rst:404]:
```suggestion
    -   That custom :class:`~scrapy.Request` object is now sent as the
```

File: docs/news.rst [hunk 5]
```diff
@@ -1918,13 +1918,13 @@ The following deprecated APIs have been removed (:issue:`3578`):
...
 *   From both ``scrapy.selector`` and ``scrapy.selector.lxmlsel``:
 
-    *   ``HtmlXPathSelector`` (use :class:`~scrapy.selector.Selector`)
+    *   ``HtmlXPathSelector`` (use :class:`~scrapy.Selector`)
 
-    *   ``XmlXPathSelector`` (use :class:`~scrapy.selector.Selector`)
+    *   ``XmlXPathSelector`` (use :class:`~scrapy.Selector`)
 
-    *   ``XPathSelector`` (use :class:`~scrapy.selector.Selector`)
+    *   ``XPathSelector`` (use :class:`~scrapy..Selector`)
```
Comment (elacuesta) [inline docs/news.rst:1925]:
```suggestion
    *   ``XPathSelector`` (use :class:`~scrapy.Selector`)
```

Comment (mmitropoulou) [reply docs/news.rst]: Thank you for pointing out these errors. If you find anything else please let me know.

**Review by Gallaecio — COMMENTED** [review]
We need to be careful when editing `news.rst`. Unlike other pages in the documentation, this page covers past versions where the code (e.g. imports available) might have been different. For example, it [seems](https://github.com/scrapy/scrapy/blob/0.8/scrapy/__init__.py) there was no `scrapy.Item` in 0.8, so the last change in `news.rst` is definitely wrong.

Since one of the goals of this change is to keep all import links working, I would suggest not changing the `news.rst` page at all.

**Comment by Gallaecio** [comment]
The changes look good to me. I think we can merge as soon as we address https://github.com/scrapy/scrapy/pull/5099#discussion_r620344448 and the issue reported by CI (sounds like `.. class:: scrapy.Item` may be used more than once now, if so we need to remove one of those usages).

**Comment by Gallaecio** [comment]
@marlenachatzigrigoriou I must say I’m not sure myself how to do it. I suggest you try https://stackoverflow.com/a/21014534/939364. If that does not work either, let me know, I’ll give it a try myself.

**Comment by marlenachatzigrigoriou** [comment]
@Gallaecio I think the way you suggested is not for this case. " _for the commits you haven’t pushed yet_" and I've already committed and pushed the changes in commits that contain more than one file.
`git checkout <commit_hash> <file_name>` is the most proper way, I found, to do it, but the lines-ending is still wrong.

**Comment by Gallaecio** [comment]
I’ll give it a try when I get a chance.

**Comment by mmitropoulou** [comment]
Hello @Gallaecio!  @marlenachatzigrigoriou and I have fixed the lines ending problem in spiders.rst & practices.rst. It would be nice if you could check the files, so the pull request may hopefully get merged. Thank you!

File: docs/topics/practices.rst [hunk 6]
```diff
@@ -119,7 +119,6 @@ Here is an example that runs multiple spiders simultaneously:
 
     import scrapy
     from scrapy.crawler import CrawlerProcess
-    from scrapy.utils.project import get_project_settings
```
Comment (elacuesta) [inline docs/topics/practices.rst:122]: Changes in this file seem unrelated to the other changes in the PR about import paths. Are they necessary? Could they be taken out to a different PR?

Comment (marlenachatzigrigoriou) [reply docs/topics/practices.rst:122]: These changes were dragged in this PR because of [427bf73](https://github.com/scrapy/scrapy/pull/5099/commits/427bf73850ad79a36a2fe7c549246f2d9f57be2f), which is one of the commits that had occurred due to a rebase. 

Indeed, they are out of this PR scope. Could we remove these lines? I guess they already belong to another PR (as they are included in the [practices.rst](https://github.com/scrapy/scrapy/blob/master/docs/topics/practices.rst) file of scrapy master branch).

Comment (mmitropoulou) [reply docs/topics/practices.rst]: Hello @elacuesta.These changes are out of the  PR scope and as @marlenachatzigrigoriou is pointing out, happend due to the above two commites (427bf73 and e66fbc4) .
We have now fixed it and in "files changed" there are no changes in the practices file.

Comment (elacuesta) [reply docs/topics/practices.rst]: Thanks for the update!

File: docs/topics/items.rst [hunk 7]
```diff
@@ -199,7 +200,12 @@ It's important to note that the :class:`Field` objects used to declare the item
 do not stay assigned as class attributes. Instead, they can be accessed through
 the :attr:`Item.fields` attribute.
 
+<<<<<<< HEAD
```
Comment (wRAR) [inline docs/topics/items.rst:203]: A conflict is committed here.

File: docs/topics/spiders.rst [hunk 8]
```diff
@@ -50,6 +50,7 @@ We will talk about those types here.
 scrapy.Spider
 =============
 
+.. class:: scrapy.spiders.Spider()
 .. class:: Spider()
```
Comment (Gallaecio) [inline docs/topics/spiders.rst:54]:
```suggestion
.. class:: scrapy.Spider()
```

(the issue was already there, so not a must)

**Review by Gallaecio — APPROVED** [review]
Looks great to me, thank you for such a thorough change.

**Review by wRAR — APPROVED** [review]
